components/datetime/src/format/timezone.rs

```diff
@@ -0,0 +1,227 @@
+                }
+            },
+            fields::TimeZone::LowerX => match u8::from(field.length) {
+                1 => time_zone_format.iso8601_format(
+                    time_zone,
+                    IsoFormat::UtcBasic,
+                    IsoMinutes::Optional,
+                    IsoSeconds::Never,
+                ),
+                2 => time_zone_format.iso8601_format(
+                    time_zone,
+                    IsoFormat::UtcBasic,
+                    IsoMinutes::Required,
+                    IsoSeconds::Never,
+                ),
+                3 => time_zone_format.iso8601_format(
+                    time_zone,
+                    IsoFormat::UtcExtended,
+                    IsoMinutes::Required,
+                    IsoSeconds::Never,
+                ),
+                4 => time_zone_format.iso8601_format(
+                    time_zone,
+                    IsoFormat::UtcBasic,
+                    IsoMinutes::Required,
+                    IsoSeconds::Optional,
+                ),
+                5 => time_zone_format.iso8601_format(
+                    time_zone,
+                    IsoFormat::UtcExtended,
+                    IsoMinutes::Required,
+                    IsoSeconds::Optional,
+                ),
+                _ => {
+                    return Err(Error::Pattern(PatternError::FieldTooLong(
+                        FieldSymbol::TimeZone(zone_symbol),
+                    )))
+                }
+            },
+            fields::TimeZone::UpperX => match u8::from(field.length) {
+                1 => time_zone_format.iso8601_format(
+                    time_zone,
+                    IsoFormat::Basic,
+                    IsoMinutes::Optional,
+                    IsoSeconds::Never,
+                ),
+                2 => time_zone_format.iso8601_format(
+                    time_zone,
+                    IsoFormat::Basic,
+                    IsoMinutes::Required,
+                    IsoSeconds::Never,
+                ),
+                3 => time_zone_format.iso8601_format(
+                    time_zone,
+                    IsoFormat::Extended,
+                    IsoMinutes::Required,
+                    IsoSeconds::Never,
+                ),
+                4 => time_zone_format.iso8601_format(
+                    time_zone,
+                    IsoFormat::Basic,
+                    IsoMinutes::Required,
+                    IsoSeconds::Optional,
+                ),
+                5 => time_zone_format.iso8601_format(
+                    time_zone,
+                    IsoFormat::Extended,
+                    IsoMinutes::Required,
+                    IsoSeconds::Optional,
+                ),
+                _ => {
+                    return Err(Error::Pattern(PatternError::FieldTooLong(
+                        FieldSymbol::TimeZone(zone_symbol),
+                    )))
+                }
+            },
+        };
+        w.write_str(&s)?;
```

Comment:
Suggestion: `s` is a `Cow<str>`, but you don't do anything with `s` except append it to `w`.  Therefore, it's wasteful for the helper functions like localized_gmt_format (and others) to have to allocate a string.  It would be better if you passed `&mut w` as an argument to those functions instead.